features/zoeken-en-raadplegen/met-burgerservicenummer/fout-cases.feature

```diff
@@ -0,0 +1,107 @@
+
+  @fout-case
+  Scenario: De burgerservicenummer parameter is niet opgegeven
+    Als personen wordt gezocht met de volgende parameters
+    | naam   | waarde                          |
+    | type   | RaadpleegMetBurgerservicenummer |
+    | fields | burgerservicenummer             |
+    Dan heeft de response een object met de volgende gegevens
+    | naam     | waarde                                                      |
+    | type     | https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.1 |
+    | title    | Minimale combinatie van parameters moet worden opgegeven.   |
+    | status   | 400                                                         |
+    | detail   | De foutieve parameter(s) zijn: burgerservicenummer.         |
+    | code     | paramsCombination                                           |
+    | instance | /haalcentraal/api/brp/personen                              |
+    En heeft het object de volgende 'invalidParams' gegevens
+    | code     | name                | reason                  |
+    | required | burgerservicenummer | Parameter is verplicht. |
+
+  @fout-case
+  Abstract Scenario: <titel>
+    Als personen wordt gezocht met de volgende parameters
+    | naam                | waarde                          |
+    | type                | RaadpleegMetBurgerservicenummer |
+    | burgerservicenummer | <burgerservicenummers>          |
+    | fields              | burgerservicenummer             |
+    Dan heeft de response een object met de volgende gegevens
+    | naam     | waarde                                                      |
+    | type     | https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.1 |
+    | title    | Een of meerdere parameters zijn niet correct.               |
+    | status   | 400                                                         |
+    | detail   | <detail>                                                    |
+    | code     | paramsValidation                                            |
+    | instance | /haalcentraal/api/brp/personen                              |
+    En heeft het object de volgende 'invalidParams' gegevens
+    | code   | name   | reason   |
+    | <code> | <name> | <reason> |
+
+    Voorbeelden:
+    | burgerservicenummers | titel                                                                   | detail                                                 | code     | name                   | reason                                      |
+    |                      | De burgerservicenummer parameter is leeg                                | De foutieve parameter(s) zijn: burgerservicenummer.    | minItems | burgerservicenummer    | Array bevat minder dan 1 items.             |
+    | 1234567890           | De burgerservicenummer parameter bevat een ongeldig burgerservicenummer | De foutieve parameter(s) zijn: burgerservicenummer[0]. | pattern  | burgerservicenummer[0] | Waarde voldoet niet aan patroon ^[0-9]{9}$. |
```

Comment:
Een 10-cijferige code die voldoet aan het patroon ^[0-9]{9}$ is nog niet per definitie een correct bsn. Deze moet bijvoorbeeld ook nog voldoen aan een 11 proef. Dat hebben we niet in de OAS specificatie opgenomen maar betekent dat dan dat de API geen foutmelding teruggeeft als het bsn niet voldoet aan de 11-proef?
Als de API daar wel een foutmelding op teruggeeft dan:
* Lijkt mij dat we er op zijn minst bij de documentatie van het bsn in de OAS specificatie iets over moeten zeggen.
* Moet dat scenario dan niet ook hier opgenomen worden?

Reply:
In principe is het geen issue als de waarde van de bsn parameter niet voldoet aan de 11-proef. Als het niet voldoet, kan er ook geen persoon gevonden worden met de bsn. Als die maar wordt gevalideerd bij het aanmaken van een persoon ;-)
Ook zou het voor ons moeilijker maken voor het kiezen van bsn voor scenario's. 000000025 voldoet bijv. niet aan de 11-proef